### ISSUE TITLE
`useless_conversion` when zipping `vec`s not triggered in `async fn`

### ISSUE THREAD

**Issue by Sufflope** [rust-lang/rust-clippy#16590]
### Summary

Zipping two vecs and calling `into_iter` on the second one only triggers the lint if the function is sync.

### Lint Name

useless_conversion

### Reproducer

Given this code:

```rust
pub fn foo() {
    let a: Vec<usize> = vec![];
    let b: Vec<usize> = vec![];

    let _ = a.into_iter().zip(b.into_iter());
}

pub async fn bar() {
    let a: Vec<usize> = vec![];
    let b: Vec<usize> = vec![];

    let _ = a.into_iter().zip(b.into_iter());
}
```

the lint is only triggered in the sync function:

```shell
cargo clippy
warning: explicit call to `.into_iter()` in function argument accepting `IntoIterator`
   --> src/lib.rs:5:31
    |
  5 |     let _ = a.into_iter().zip(b.into_iter());
    |                               ^------------
    |                                |
    |                                help: consider removing the `.into_iter()`
    |
note: this parameter accepts any `IntoIterator`, so you don't need to call `.into_iter()`
   --> /home/foo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:616:12
    |
616 |         U: IntoIterator,
    |            ^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#useless_conversion
    = note: `#[warn(clippy::useless_conversion)]` on by default

warning: `foo` (lib) generated 1 warning (run `cargo clippy --fix --lib -p foo` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.00s
```

### Version

```text
rustc 1.93.1 (01f6ddf75 2026-02-11)
binary: rustc
commit-hash: 01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf
commit-date: 2026-02-11
host: x86_64-unknown-linux-gnu
release: 1.93.1
LLVM version: 21.1.8
```
